### PR TITLE
Count sim timer down rather than up

### PR DESCRIPTION
**Pull request by ErikBoesen**
`getMatchTime()` should return the number of seconds remaining in the simulated match, not the number of seconds for which the match has been running.

### CODE REVIEW

File: hal-sim/hal_impl/functions.py [hunk 1]
```diff
@@ -1056,7 +1056,7 @@ def getMatchTime(status):
     status.value = 0
     match_start = hal_data['time']['match_start']
     if match_start is None:
-        return 0.0
+        return 135.0
     else:
         return (hooks.getFPGATime() - hal_data['time']['match_start'])/1000000.0
```
Comment (twinters007) [inline hal-sim/hal_impl/functions.py]: This should be swapped.

**Comment by ErikBoesen** [comment]
I've resolved the issues pointed out by @twinters007.

File: hal-sim/hal_impl/mode_helpers.py [hunk 2]
```diff
@@ -5,114 +5,114 @@
...
         'autonomous': False,
         'test': False,
         'enabled': enabled,
         'dsAttached': True
     })
-    
+
     if enabled:
-        hal_data['time']['match_start'] = fns.hooks.getFPGATime() - 15000000
+        hal_data['time']['match_start'] = 135000000 - 15000000
```
Comment (twinters007) [inline hal-sim/hal_impl/mode_helpers.py]: I think we should change the name of this key to `time_remaining` and currently, if I start the sim, wait 20 seconds, then enable again, the time won't be at the beginning of teleop, instead it'll say there are 100 seconds left. I don't think this is desired behavior. I think you should do `fns.hooks.getFPGATime() + 120000000`

Comment (ErikBoesen) [reply hal-sim/hal_impl/mode_helpers.py]: I feel like just calling it `remaining` may be preferable, given that `time` is used in the parent dictionary.

**Comment by ErikBoesen** [comment]
Fixed, with the caveat of calling the value `remaining` instead.

File: hal-sim/hal_impl/functions.py [hunk 3]
```diff
@@ -1054,11 +1054,11 @@ def getMatchTime(status):
...
     status.value = 0
-    match_start = hal_data['time']['match_start']
-    if match_start is None:
-        return 0.0
+    remaining = hal_data['time']['remaining']
+    if remaining is None:
+        return 135.0
     else:
-        return (hooks.getFPGATime() - hal_data['time']['match_start'])/1000000.0
+        return (hal_data['time']['remaining'] - hooks.getFPGATime())/1000000.0
```
Comment (twinters007) [inline hal-sim/hal_impl/functions.py]: Why do the dict lookup twice?

File: hal-sim/hal_impl/mode_helpers.py [hunk 4]
```diff
@@ -5,114 +5,114 @@
...
         'autonomous': False,
         'test': False,
         'enabled': enabled,
         'dsAttached': True
     })
-    
+
     if enabled:
-        hal_data['time']['match_start'] = fns.hooks.getFPGATime() - 15000000
+        hal_data['time']['remaining'] = hal_data['time']['remaining'] + 12000000
```
Comment (twinters007) [inline hal-sim/hal_impl/mode_helpers.py]: missing a 0. And if I enable autonomous (remaining = 135) then enable teleop, remaining now equals 255. Add to getFPGATime instead

**Pull request merged**